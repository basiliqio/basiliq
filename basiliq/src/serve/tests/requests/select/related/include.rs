use super::*;

crate::run_test_request!(
    include_rel_m2o,
    Method::GET,
    format!(
        "/public__articles/{}/public__peoples?include=public__favorite_color",
        ARTICLES_IDS[0]
    ),
    200
);

crate::run_test_request!(
    include_self,
    Method::GET,
    format!(
        "/public__peoples/{}/public__articles?include=public__peoples",
        PEOPLES_IDS[0]
    ),
    200
);

crate::run_test_request!(
    include_rel_o2m,
    Method::GET,
    format!(
        "/public__peoples/{}/public__articles?include=public__people-article",
        PEOPLES_IDS[0]
    ),
    200
);

crate::run_test_request!(
    include_distant,
    Method::GET,
    format!(
        "/public__peoples/{}/public__articles?include=public__peoples.public__comments",
        PEOPLES_IDS[0]
    ),
    200
);
crate::run_test_request!(
    include_distant_self,
    Method::GET,
    format!(
        "/public__peoples/{}/public__articles?include=public__peoples.public__comments.public__articles",
        PEOPLES_IDS[0]
    ),
    200
);
