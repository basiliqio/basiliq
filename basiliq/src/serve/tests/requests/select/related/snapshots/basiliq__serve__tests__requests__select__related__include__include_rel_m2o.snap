---
source: basiliq/src/serve/tests/requests/select/related/include.rs
expression: res

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": [
    {
      "type": "public__peoples",
      "id": "1649b1e9-8a5f-4f52-b331-c07ce3bccc6f",
      "attributes": {
        "age": 22,
        "first-name": "Francis",
        "gender": "M",
        "last-name": "Le Roy",
        "twitter": null
      },
      "relationships": {
        "public__favorite_color": {
          "data": {
            "type": "public__favorite_color",
            "id": "9302b136-69ef-4f0d-b907-10593ff7e578"
          }
        }
      }
    }
  ],
  "included": [
    {
      "type": "public__favorite_color",
      "id": "9302b136-69ef-4f0d-b907-10593ff7e578",
      "attributes": {
        "color": "red"
      }
    }
  ]
}
